pkg/api/v1/resource/helpers.go

```diff
@@ -65,37 +68,75 @@ func PodRequestsAndLimits(pod *v1.Pod) (reqs, limits v1.ResourceList) {
 		maxResourceList(reqs, container.Resources.Requests)
 		maxResourceList(limits, container.Resources.Limits)
 	}
+
+	// if PodOverhead feature is supported, add overhead for running a pod
+	// to the non-zero sum of container requests/limits:
+	if pod.Spec.Overhead != nil && utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead) {
+		applyPodOverhead(pod.Spec.Overhead, reqs)
+		applyPodOverhead(pod.Spec.Overhead, limits)
+	}
+
 	return
 }
 
-// GetResourceRequest finds and returns the request for a specific resource.
-func GetResourceRequest(pod *v1.Pod, resource v1.ResourceName) int64 {
-	if resource == v1.ResourcePods {
-		return 1
+// applyPodOverhead will apply overheads to applicable resources which are non zero
+func applyPodOverhead(overhead v1.ResourceList, resources v1.ResourceList) {
+	for name, quantity := range overhead {
+		if value, ok := resources[name]; ok && !value.IsZero() {
+			value.Add(quantity)
+			resources[name] = value
+		}
+	}
+}
+
+// GetResourceRequestQuantity finds and returns the request quantity for a specific resource.
+func GetResourceRequestQuantity(pod *v1.Pod, resourceName v1.ResourceName) resource.Quantity {
+
+	requestQuantity := resource.Quantity{Format: resource.BinarySI}
```

Comment:
Can we switch the Format based on resource, and default to `resource.DecimalSI` if it is a resource that it isn't in types/v1?  It doesn't matter for eviction, but will be useful if anyone else uses the function.

Reply:
ACK.  

FYI: opened this issue, as I see Format being set inconsistently (in tests only): https://github.com/kubernetes/kubernetes/issues/81379

Reply:
addressed